src/server/detail/wrapped_json_path.h

```diff
@@ -248,38 +232,23 @@ class WrappedJsonPath {
     return mutate_result;
   }
 
-  bool IsLegacyModePath() const {
-    return path_type_ == JsonPathType::kLegacy;
-  }
+  bool IsLegacyModePath() const;
 
-  bool RefersToRootElement() const {
-    auto path = path_.view();
-    return path.empty() || path == kV1PathRootElement || path == kV2PathRootElement;
-  }
+  bool RefersToRootElement() const;
 
-  bool HoldsJsonPath() const {
-    return std::holds_alternative<json::Path>(parsed_path_);
-  }
+  // Returns true if this is first implementation of json path
+  bool HoldsJsonPath() const;
 
-  const json::Path& AsJsonPath() const {
-    return std::get<json::Path>(parsed_path_);
-  }
+  // Main implementation of json path
```

Comment:
"main" and "alternative" are not clear here.  jsoncons and internal are better terms